### PR TITLE
[KNX2] Add Color Channel - KNX2_addColorChannel_rev3

### PR DESCRIPTION
**Pull request by lewie**
*CHANGES*
KNXCoreTypeMapper: Added percentage manipulation
KNXChannelType: PATTERN can deal with 123.1234 now 
KNXChannelType, AbstractKNXClient: Added some traces for better debug.

*DESCRIPTION*
Solution for the bugs from: https://github.com/openhab/openhab2-addons/pull/3433

*TESTING*
Disable knx in PaperUI or /etc/openhab2/services/addons.cfg and copy [org.openhab.binding.knx-2.4.0-SNAPSHOT.jar](https://github.com/lewie/openhab2/raw/KNX2_addColorChannel_rev3_testfile/org.openhab.binding.knx-2.4.0-SNAPSHOT.jar) to /usr/share/openhab2/addons

### CODE REVIEW

**Comment by FredericMa** [comment]
Fixes my issue when selecting a color!

**Comment by kaikreuzer** [comment]
You only missed to sign-off your commit 😟

**Comment by lewie** [comment]
@kaikreuzer,
Thank You! 😬
I have signed both now.

File: addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/channel/KNXChannelType.java [hunk 1]
```diff
@@ -157,6 +157,7 @@ protected final boolean isEquals(@Nullable Configuration configuration, String a
 
     public final @Nullable OutboundSpec getCommandSpec(Configuration configuration, KNXTypeMapper typeHelper,
             Type command) throws KNXFormatException {
+        logger.trace("getCommandSpec testing Keys '{}' for command '{}'", getAllGAKeys(), command);
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/channel/KNXChannelType.java]: @lewie Just a (important?) question how heavy is the `getAllGAKeys()` method, it is called an extra time even when the logging level is non-trace is this okay in this case?

Comment (lewie) [reply addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/channel/KNXChannelType.java]: getAllGAKeys means normally one or maximum 5 Keys. For logger.trace it is helpful at this position.

Comment (martinvw) [reply addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/channel/KNXChannelType.java]: Yes but does it do a calculation or is it a more less static call?

Comment (lewie) [reply addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/channel/KNXChannelType.java]: It is static:
```
    @Override
    protected Set<String> getAllGAKeys() {
        return Stream.of(SWITCH_GA, POSITION_GA, INCREASE_DECREASE_GA, HSB_GA).collect(toSet());
    }
```

Comment (martinvw) [reply addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/channel/KNXChannelType.java]: So my point is indeed that it now will always construct that set twice (also when running with info logging) so maybe a next time you could fix the code that it does collect the set only once.

Comment (lewie) [reply addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/channel/KNXChannelType.java]: Yes, this makes sense.

**Pull request merged**